NHLBI_BioData_Catalyst/python/5_LongitudinalData.ipynb

```diff
@@ -34,476 +34,530 @@
-    "lipid_vars = pd.concat([lipidvariablesDict, triglyceridevariablesDict])\n",
-    "\n",
-    "lipid_vars"
+    "filtered_lipid_dataframe = lipid_dataframe[lipid_dataframe.studyId.str.contains('phs000007')]\n",
+    "filtered_lipid_dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "99a2aa80",
+   "metadata": {},
+   "source": [
+    "As you can see, there are a number of variables in the Framingham Heart Study which are related to lipids or triglicerides. In this case study, we are interested specifically in `longitudinal` data, or variables which have been measured over time. "
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "25da3905",
+   "id": "f7984edf",
    "metadata": {},
    "source": [
     "### Identify the longitudinal lipid variables\n",
-    "The following code does the following:\n",
-    "\n",
-    "- uses the dataframe containing variables which are related to 'lipid' or 'triglyceride'\n",
-    "- filters for variables with keywords 'exam #' or 'visit #'\n",
-    "- extracts the exam number of each variable into column `exam_number`\n",
-    "- groups variables longitudinal variable (`longvar`)\n",
-    "- returns a table showing the variables that have more than one exam recorded"
+    "In order to identify which lipid variables are measured over time, we will take advantage of the keywords `exam` and `visit`. Through a brief review of our lipid variables in the Framingham Heart Study, we can see that many variables contain an exam or visit number, indicating that it is longitudinal data.\n"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "4dfdf3fa",
+   "id": "a58b1890",
    "metadata": {},
    "source": [
-    "First, the longitudinal concept paths that have exam or visit number information will be saved to `longitudinal_concept_paths`."
+    "First, we will filter our dataframe containing variables which are related to `lipid` or `triglyceride` in Framingham Heart Study to those which have the keywords `exam #` or `visit #`."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "afbab004",
+   "id": "7ef1e0c6",
    "metadata": {},
    "outputs": [],
    "source": [
-    "longitudinal_concept_paths = [] # Initialize empty list\n",
-    "for i in lipid_vars['name']:\n",
-    "    # If variable has exam # or visit #, add it to longitudinal_concept_paths\n",
-    "    if re.search('(exam \\d+|visit \\d+)', i, re.IGNORECASE):\n",
-    "        longitudinal_concept_paths.append(i)\n",
-    "len(longitudinal_concept_paths)"
+    "filtered_lipid_dataframe = filtered_lipid_dataframe[filtered_lipid_dataframe.description.str.contains('exam|visit', case = False)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5f714e7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['description'].str.extract('(exam \\d+|visit \\d+)', flags = re.IGNORECASE)[0]"
```

Comment:
This cell works, but gives a warning:

/tmp/ipykernel_1167/3898402110.py:1: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
filtered_lipid_dataframe['exam_number'] = filtered_lipid_dataframe['description'].str.extract('(exam \d+|visit \d+)', flags = re.IGNORECASE)[0]